src/box/space_def.c

```diff
@@ -223,6 +223,8 @@ space_def_new_ephemeral(uint32_t exact_field_count, struct field_def *fields)
 void
 space_def_delete(struct space_def *def)
 {
+	if (def == NULL)
+		return;
```

Comment:
I don't like that this function handles `def = NULL` while, for example, `index_def_delete` or `key_def_delete`. Please move this check to where this function is called.

Reply:
Fixed.

<details>
<summary>diff</summary>

```diff
diff --git a/src/box/alter.cc b/src/box/alter.cc
index b413fe42e3..a28d750ee7 100644
--- a/src/box/alter.cc
+++ b/src/box/alter.cc
@@ -2116,8 +2116,10 @@ on_replace_dd_space(struct trigger * /* trigger */, void *even>
                        ER_CREATE_SPACE : ER_ALTER_SPACE;
                def = space_def_new_from_tuple(new_tuple, errcode, region);
        }
-       auto def_guard =
-               make_scoped_guard([=] { space_def_delete(def); });
+       auto def_guard = make_scoped_guard([=] {
+               if (def != NULL)
+                       space_def_delete(def);
+       });
        if (filter_temporary_ddl_stmt(txn, old_space != NULL ?
                                      old_space->def : def) != 0)
                return -1;
diff --git a/src/box/space_def.c b/src/box/space_def.c
index 945d4c31c3..3dd9da133b 100644
--- a/src/box/space_def.c
+++ b/src/box/space_def.c
@@ -223,8 +223,6 @@ space_def_new_ephemeral(uint32_t exact_field_count, struct field_>
 void
 space_def_delete(struct space_def *def)
 {
-       if (def == NULL)
-               return;
        field_def_array_delete(def->fields, def->field_count);
        tuple_dictionary_unref(def->dict);
        free(def->opts.sql);
```

</details>